integration-cli/docker_api_swarm_test.go

```diff
@@ -445,6 +445,43 @@ func (s *DockerSwarmSuite) TestApiSwarmServicesStateReporting(c *check.C) {
 		}
 	}
 }
+func (s *DockerSwarmSuite) TestApiSwarmLeaderElection(c *check.C) {
+	// Create 3 nodes
+	d1 := s.AddDaemon(c, true, true)
+	d2 := s.AddDaemon(c, true, true)
+	d3 := s.AddDaemon(c, true, true)
+
+	// pick out the leader
+	var leader *SwarmDaemon
+	for _, d := range []*SwarmDaemon{d1, d2, d3} {
+		if d1.getNode(c, d.NodeID).ManagerStatus.Leader {
+			leader = d
+			break
+		}
+	}
+
+	// stop the leader
+	leader.Stop()
+
+	// wait for an election to occur
+	time.Sleep(time.Second * 2)
```

Comment:
Seems like we should be able to loop on waiting for a leader than a `time.Sleep`.
`time.Sleep` is going to be racey, as well as always delay the test for the full length of the sleep.